app/controllers/catalog_controller.rb

```diff
@@ -261,6 +261,28 @@ def atomic_form_field_changed
 
     changed = (@edit[:new] != @edit[:current])
 
+    if changed && @edit[:new][:ovf_template_id]
+      ovf_template = ManageIQ::Providers::Vmware::InfraManager::OrchestrationTemplate.find_by(:id => @edit[:new][:ovf_template_id])
+      @edit[:new][:src_vm_id] = @edit[:new][:ovf_template_id]
+      ovf_template.refresh_field_values(@edit[:new])
+
+      @edit[:available_folders] = ovf_template.allowed_folders
+                                              .collect { |m| [m.last, m.first] }
+                                              .sort
+      @edit[:available_resource_pools] = ovf_template.allowed_resource_pools
+                                                     .collect { |m| [m.last, m.first] }
+                                                     .sort
+      @edit[:available_datacenters] = ovf_template.allowed_datacenters
+                                                  .collect { |m| [m.last, m.first] }
+                                                  .sort
+      @edit[:available_hosts] = ovf_template.allowed_hosts
```

Comment:
Strange that allowed_hosts seems to return the host objects, but the other allowed methods above don't?

Reply:
yeah I'm not really sure why that is, I just copied it from `fetch_form_vars_ovf_template`.

Reply:
@Fryguy ^

Reply:
yeah, that's fine - just wanted to raise it up as something we may need to clean up in the future.